base/applications/calc/winmain.c

```diff
@@ -637,7 +640,12 @@ static void build_operand(HWND hwnd, DWORD idc)
             return;
         break;
     }
-    calc.ptr += _stprintf(calc.ptr, _T("%C"), key2code[i].key);
+
+    cbPtr = sizeof(calc.buffer) - ((BYTE*)calc.ptr - (BYTE*)calc.buffer);
+
+    StringCbPrintf(calc.ptr, cbPtr, _T("%C"), key2code[i].key);
+
+    calc.ptr += _tcslen(calc.ptr);
```

Comment:
I wonder whether some of this stuff can be managed by using StringCbPrintfEx instead.

Reply:
Fixed.